engine/schema/src/com/cloud/host/dao/HostDaoImpl.java

```diff
@@ -1192,6 +1193,15 @@ public HostVO findHostInZoneToExecuteCommand(long zoneId, HypervisorType hypervi
         }
     }
 
+    @Override
+    public List<HostVO> listAllHostsUpByZoneAndHypervisor(long zoneId, HypervisorType hypervisorType) {
+        return listByDataCenterIdAndHypervisorType(zoneId, hypervisorType)
+                .stream()
```

Comment:
Why not do the filtering directly in the SQL/Select? Then, we would not need to load/create objects just to discard them here.